src/main/kotlin/net/atos/zac/app/informatieobjecten/EnkelvoudigInformatieObjectRestService.kt

```diff
@@ -296,32 +297,27 @@ class EnkelvoudigInformatieObjectRestService @Inject constructor(
 
     @GET
     @Path("informatieobjecttypes/zaak/{zaakUuid}")
-    fun listInformatieobjecttypesForZaak(@PathParam("zaakUuid") zaakID: UUID): List<RESTInformatieobjecttype> {
-        val zaak = zrcClientService.readZaak(zaakID)
-        val zaaktype = ztcClientService.readZaaktype(zaak.zaaktype)
-        val informatieObjectTypes = zaaktype.informatieobjecttypen
+    fun listInformatieobjecttypesForZaak(@PathParam("zaakUuid") zaakID: UUID): List<RESTInformatieobjecttype> =
+        zrcClientService.readZaak(zaakID).zaaktype
+            .let { ztcClientService.readZaaktype(it).informatieobjecttypen }
             .map { ztcClientService.readInformatieobjecttype(it) }
-            .filter { InformatieObjectTypeUtil.isNuGeldig(it) }
+            .filter { isNuGeldig(it) }
             .toList()
-        return informatieObjecttypeConverter.convert(informatieObjectTypes)
-    }
+            .let(informatieObjecttypeConverter::convert)
 
     @GET
     @Path("zaakinformatieobject/{uuid}/informatieobject")
     fun readEnkelvoudigInformatieobjectByZaakInformatieobjectUUID(
-        @PathParam("uuid") uuid: UUID?
-    ): RESTEnkelvoudigInformatieobject = informatieobjectConverter.convertToREST(
-        drcClientService.readEnkelvoudigInformatieobject(
-            zrcClientService.readZaakinformatieobject(uuid).informatieobject
-        )
-    )
+        @PathParam("uuid") uuid: UUID
+    ): RESTEnkelvoudigInformatieobject =
+        zrcClientService.readZaakinformatieobject(uuid).informatieobject
+            .let(drcClientService::readEnkelvoudigInformatieobject)
+            .let(informatieobjectConverter::convertToREST)
 
     @GET
     @Path("informatieobject/{uuid}/zaakinformatieobjecten")
-    fun listZaakInformatieobjecten(@PathParam("uuid") uuid: UUID?): List<RESTZaakInformatieobject> =
-        drcClientService.readEnkelvoudigInformatieobject(
-            uuid
-        ).let {
+    fun listZaakInformatieobjecten(@PathParam("uuid") uuid: UUID): List<RESTZaakInformatieobject> =
+        drcClientService.readEnkelvoudigInformatieobject(uuid).let {
             assertPolicy(policyService.readDocumentRechten(it).lezen)
             return zrcClientService.listZaakinformatieobjecten(it)
```

Comment:
looks better indeed. thanks!

Reply:
oh yeah, tried that already but `drcClientService::readEnkelvoudigInformatieobject` does not work because it is an overloaded Java method. That is why I stuck with the return in the `let`. why would that not make any sense btw? maybe not pretty but it works fine. I propose to leave it like this until we convert the `drcClientService` to Kotlin and can improve that code.